src/trigger.jl

```diff
@@ -33,13 +33,13 @@ LegendGeSimConfig -> TrapFilter
 
 Construct a TrapFilter instance based on simulation configuration given in <sim_conf>.
 """
-function TrapFilter(sim_conf::LegendGeSimConfig)
+function TrapFilter(trigger_settings::PropDict)
     T = Float32 # This should be somehow defined and be passed properly
     TrapFilter(
-        window_lengths = (sim_conf.dict.setup.trigger.window_lengths[1],
-            sim_conf.dict.setup.trigger.window_lengths[2],
-            sim_conf.dict.setup.trigger.window_lengths[3]),
-        threshold_keV = T(sim_conf.dict.setup.trigger.threshold)u"keV"
+        window_lengths = (trigger_settings.window_lengths[1],
+            trigger_settings.window_lengths[2],
+            trigger_settings.window_lengths[3]),
+        threshold_keV = T(trigger_settings.threshold)u"keV"
     )
 end
```

Comment:
Will `threshold` be determined from `threshold_keV` and if so, is this already in the code somewhere?

Reply:
Yes, for now it's in `temp_utils.jl`, and we convert keV with Ge ionization E and preamp gain in case of `NoiseFromSim`; and if data baselines are used there's also an option to NOT give threshold in keV, then 3 sigma from baseline is used.

This was like just dummy design though. I have no idea how it works in real DAQ, how do we set the threshold? Do we set it in keV? Or ADC? Or relative e.g. N sigma from baseline? The threshold in keV here was some very first thing @oschulz suggested 2 years ago probably just to have something.

Reply:
We should take this `TrapFilter` here out and use the one from RadiationDetectorDSP

Reply:
But that can be done later.

Reply:
Yes I think first I'd merge this, and then I'll put stuff in Issues and we can assign who can do what. TrapFilter is top priority because it slows everything down (I mean in my current for loop form...)